frontend/packages/dev-console/src/components/import/advanced/ResourceLimitSection.tsx

```diff
@@ -12,7 +15,7 @@ const ResourceLimitSection: React.FC = () => {
         label="Request"
         unitName="limits.cpu.requestUnit"
         unitOptions={CPUUnits}
-        defaultUnitSize="m"
+        defaultUnitSize={`${cpuLimits.value.requestUnit}`}
```

Comment:
Oh boy this makes me uncomfortable. `default` based props are not supposed to change. I followed it down into the underlying component and it feels wrong there too... so I am unsure what to do here. I don't think this change is needed for it to work but I haven't fully tested.

@divyanshiGupta if this change was to fix the storing of the correct information then please keep it. If it is not a needed change, I think we shouldn't do it... but overall I don't think it's super important one way or another. It just feels wrong using a `default` prop as the live update value.

Reply:
I made this change to show the unit size that was selected when the application was created as the default. I can create a separate prop in limits section for the default value which will be only initialized once and will not update with selecting another unit from the dropdown.